File: ServiceTest/app/src/main/java/com/example/servicetest/ExampleService.java
package com.example.servicetest;

import android.app.Notification;
import android.app.PendingIntent;
import android.app.Service;
import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.content.IntentFilter;
import android.os.IBinder;
import android.util.Log;
import android.widget.EditText;
import android.widget.Toast;

import androidx.annotation.Nullable;
import androidx.core.app.NotificationCompat;

import static android.net.wifi.WifiManager.NETWORK_STATE_CHANGED_ACTION;
import static com.example.servicetest.App.CHANNEL_ID;

public class ExampleService extends Service {
    BroadcastReceiver mReceiver;

    // use this as an inner class like here or as a top-level class
    public class MyReceiver extends BroadcastReceiver {

        @Override
        public void onReceive(Context context, Intent intent) {
            String action = intent.getAction();
            if(action == null){
                Log.e("error", "action is null!");
            }
            else {
                switch (action) {
                    case NETWORK_STATE_CHANGED_ACTION:
                        Log.e("found something", "wifi changed!");
                        Context c = getApplicationContext();
                        CharSequence text = "wifi changed!";
                        int duration = Toast.LENGTH_SHORT;

                        Toast toast = Toast.makeText(c, text, duration);
                        toast.show();
                }
            }

        }

        // constructor
        public MyReceiver(){

        }
    }

    @Override
    public void onCreate() {
        // get an instance of the receiver in your service
        IntentFilter filter = new IntentFilter();
        filter.addAction(NETWORK_STATE_CHANGED_ACTION);
//        filter.addAction("anotherAction");
        mReceiver = new MyReceiver();
        registerReceiver(mReceiver, filter);
    }

    @Override
    public int onStartCommand(Intent intent, int flags, int startId) {
        String input = intent.getStringExtra("inputExtra");

        Intent notificationIntent = new Intent(this, MainActivity.class);
        PendingIntent pendingIntent = PendingIntent.getActivity(this, 0,
                                                                    notificationIntent, 0);

        Notification notification = new NotificationCompat.Builder(this, CHANNEL_ID)
                .setContentTitle("Example Service")
                . setContentText(input)
                .setSmallIcon(R.drawable.ic_android)
                .setContentIntent(pendingIntent)
                .build();

        startForeground(1, notification);

        return START_STICKY;

        //stopService();
    }

    @Nullable
    @Override
    public IBinder onBind(Intent intent) {
        return null;
    }
}

File: ServiceTest/app/src/main/java/com/example/servicetest/ExampleService.kt
package com.example.servicetest

import android.app.PendingIntent
import android.app.Service
import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.content.IntentFilter
import android.net.wifi.WifiManager
import android.os.IBinder
import android.util.Log
import android.widget.Toast
import androidx.core.app.NotificationCompat

class ExampleService : Service() {
    var mReceiver: BroadcastReceiver? = null

    // use this as an inner class like here or as a top-level class
    inner class MyReceiver  // constructor
        : BroadcastReceiver() {
        override fun onReceive(context: Context, intent: Intent) {
            val action = intent.action
            if (action == null) {
                Log.e("error", "action is null!")
            } else {
                when (action) {
                    WifiManager.NETWORK_STATE_CHANGED_ACTION -> {
                        Log.e("found something", "wifi changed!")
                        val c = applicationContext
                        val text: CharSequence = "wifi changed!"
                        val duration = Toast.LENGTH_SHORT
                        val toast = Toast.makeText(c, text, duration)
                        toast.show()
                    }
                }
            }
        }
    }

    override fun onCreate() {
        // get an instance of the receiver in your service
        val filter = IntentFilter()
        filter.addAction(WifiManager.NETWORK_STATE_CHANGED_ACTION)
        //        filter.addAction("anotherAction");
        mReceiver = MyReceiver()
        registerReceiver(mReceiver, filter)
    }

    override fun onStartCommand(intent: Intent, flags: Int, startId: Int): Int {
        val input = intent.getStringExtra("inputExtra")
        val notificationIntent = Intent(this, MainActivity::class.java)
        val pendingIntent = PendingIntent.getActivity(this, 0,
                notificationIntent, 0)
        val notification = NotificationCompat.Builder(this, App.CHANNEL_ID)
                .setContentTitle("Example Service")
                .setContentText(input)
                .setSmallIcon(R.drawable.ic_android)
                .setContentIntent(pendingIntent)
                .build()
        startForeground(1, notification)
        return START_STICKY

        //stopService();
    }

    override fun onBind(intent: Intent): IBinder? {
        return null
    }
}